### PR TITLE
Exclude consumer plugins when --skip-consumers is on

### PR DESCRIPTION
**Pull request by owlwalks**
Linked issue: https://github.com/hbagdi/deck/issues/119

### CODE REVIEW

File: dump/dump.go [hunk 1]
```diff
@@ -81,6 +81,10 @@ func Get(client *kong.Client, config Config) (*utils.KongRawState, error) {
 	}
 	state.Targets = targets
 
+	if config.SkipConsumers {
```
Comment (hbagdi) [inline dump/dump.go]: nit: Please move this below line 52 to keep it near where the plugins are loaded.

File: dump/dump.go [hunk 2]
```diff
@@ -444,3 +448,15 @@ func GetAllACLGroups(client *kong.Client, tags []string) ([]*kong.ACLGroup, erro
 	}
 	return aclGroups, nil
 }
+
+// ExcludeConsumersPlugins filter out consumer plugins
+func ExcludeConsumersPlugins(plugins []*kong.Plugin) []*kong.Plugin {
```
Comment (hbagdi) [inline dump/dump.go]: Please don't export this method. Methods beginning with a capital letter are available outside the package. It makes sense to keep this method internal to the package only.

File: dump/dump.go [hunk 3]
```diff
@@ -444,3 +448,15 @@ func GetAllACLGroups(client *kong.Client, tags []string) ([]*kong.ACLGroup, erro
...
+	var filtered []*kong.Plugin
+	for _, p := range plugins {
+		if p.Consumer != nil && p.Consumer.ID != nil {
```
Comment (hbagdi) [inline dump/dump.go]: Please use `utils.Empty()` method here instead:
https://godoc.org/github.com/hbagdi/deck/utils#Empty

**Comment by hbagdi** [comment]
For the record, know one thing that this gets messy if there are plugins that are present on a combination of a consumer and route or consumer and service. There are all sorts of corner cases which decK doesn't handle today.

This change looks good though.

**Comment by owlwalks** [comment]
Yea, would be nicer if this is supported by kong admin api, I just pushed some changes regarding your comments.

**Pull request merged**